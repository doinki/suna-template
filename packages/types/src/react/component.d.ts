import type { ComponentPropsWithRef, CSSProperties, ElementType } from 'react';

export type DistributiveOmit<T, K extends keyof any> = T extends any
  ? Omit<T, K>
  : never;

export interface CommonProps {
  className?: string;
  style?: CSSProperties;
}

export interface OverridableTypeMap {
  defaultComponent: ElementType;
  props: {};
}

export type BaseProps<M extends OverridableTypeMap> = M['props'] & CommonProps;

export type OverrideProps<
  M extends OverridableTypeMap,
  C extends ElementType
> = BaseProps<M> &
  DistributiveOmit<ComponentPropsWithRef<C>, keyof BaseProps<M>>;

export type DefaultComponentProps<M extends OverridableTypeMap> = BaseProps<M> &
  DistributiveOmit<
    ComponentPropsWithRef<M['defaultComponent']>,
    keyof BaseProps<M>
  >;

export interface OverridableComponent<M extends OverridableTypeMap> {
  <C extends ElementType>(
    props: { component: C } & OverrideProps<M, C>
  ): JSX.Element | null;
  (props: DefaultComponentProps<M>): JSX.Element | null;
}
