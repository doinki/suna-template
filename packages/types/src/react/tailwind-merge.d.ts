export type ClassNameValue =
  | ClassNameValue[]
  | string
  | false
  | 0
  | null
  | undefined;
