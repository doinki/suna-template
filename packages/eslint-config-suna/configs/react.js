require('@rushstack/eslint-patch/modern-module-resolution');

const rules = {
  'react/function-component-definition': [
    'error',
    {
      namedComponents: ['arrow-function'],
      unnamedComponents: ['arrow-function'],
    },
  ],
  'react/jsx-props-no-spreading': 'off',
  'react/jsx-sort-props': [
    'warn',
    { callbacksLast: true, reservedFirst: true, shorthandLast: true },
  ],
  'react/prop-types': 'off',
  'react/require-default-props': 'off',
  'salt/react-hook-dependencies': 'warn',
  'trim/argument': 'warn',
  'trim/class-name': 'warn',
};

/**
 * @type {import('eslint').Linter.Config}
 */
module.exports = {
  env: { browser: true },
  extends: ['./base.js', 'airbnb', 'airbnb/hooks', 'plugin:react/jsx-runtime'],
  overrides: [
    {
      extends: [
        './base.js',
        'plugin:@typescript-eslint/recommended',
        'plugin:@typescript-eslint/recommended-requiring-type-checking',
        'airbnb',
        'airbnb-typescript',
        'airbnb/hooks',
        'plugin:react/jsx-runtime',
      ],
      files: '**/*.ts?(x)',
      plugins: ['typescript-sort-keys'],
      rules: {
        ...rules,
        '@typescript-eslint/consistent-type-exports': 'error',
        '@typescript-eslint/consistent-type-imports': 'error',
        'typescript-sort-keys/interface': 'warn',
      },
    },
  ],
  parserOptions: { ecmaVersion: 'latest' },
  plugins: ['salt', 'trim'],
  rules,
};
