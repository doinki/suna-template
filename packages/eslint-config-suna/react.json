{
  "$schema": "https://json.schemastore.org/eslintrc",
  "extends": [
    "./configs/react",
    "./configs/jest",
    "./configs/jest-dom",
    "./configs/testing-library",
    "./configs/storybook",
    "./configs/tailwindcss",
    "./configs/prettier",
    "./configs/custom"
  ]
}
