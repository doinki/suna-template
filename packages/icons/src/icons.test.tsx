import { render } from '@testing-library/react';

import * as icons from '.';

Object.entries(icons).forEach(([name, Icon]) => {
  describe(`<${name} />`, () => {
    test('snapshot', () => {
      const {
        container: { firstChild: icon },
      } = render(<Icon />);

      expect(icon).toMatchSnapshot();
    });
  });
});
