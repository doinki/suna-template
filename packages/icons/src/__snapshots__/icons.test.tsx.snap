// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> snapshot 1`] = `
<svg
  aria-hidden="true"
  height="1em"
  viewBox="0 0 32 32"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <lineargradient
      id="-"
    >
      <stop
        offset="0%"
        stop-color="#0f4c81"
      />
      <stop
        offset="50%"
        stop-color="#6667ab"
      />
      <stop
        offset="100%"
        stop-color="#98b4d4"
      />
    </lineargradient>
  </defs>
  <circle
    cx="16"
    cy="16"
    fill="url(#-)"
    r="14"
    stroke="#000"
    stroke-width="2"
  />
</svg>
`;
